srcTest/testFunctions.test.js

```diff
@@ -0,0 +1,10 @@
+import { search_pattern } from "./../src/Class Specs/Class_sport";
+import { test, expect } from "vitest";
+
+test("Check data to be fasle when comparing Test with TEST", () => {
+  expect(search_pattern("Test", "TEST")).toBe(false);
+});
+
+test("Expect the result to be true when searching est inside a word Called Test", () => {
+  expect(search_pattern("Test", "est")).toBe(true);
+});
```

Comment:
Try adding more test inside of these functions, one function could test all true wanted results, while the other one return all false wanted results.

Reply:
That seems like a useful thing to do. But for this type of test, it is not required-- in some way.